### PR TITLE
test(integration): re-enable some tests by renaming their filename

### PR DESCRIPTION
**Pull request by windmgc**
### Summary

It seems some of the test files are not ending with a `_spec` suffix in their file names.
Thus maybe they're not tested for a long time...

### Checklist

- [x] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* re-enable some tests by renaming their filename

### CODE REVIEW

**Comment by windmgc** [comment]
Rename the following spec files as they seems useful:
- `spec/01-unit/19-hybrid/04-negotiation_spec.lua`
- `spec/02-integration/04-admin_api/12-plugins-conf_spec.lua`
- `spec/02-integration/09-hybrid_mode/08-lazy_export_spec.lua`
- `spec/02-integration/11-dbless/03-config_persistence_spec.lua`
- `spec/03-plugins/26-prometheus/06-hybrid-mode_metrics_spec.lua`

Delete one spec file as it seems deprecated
- `spec/02-integration/04-admin_api/18-worker-events.lua` (There was an age when `worker-events` is a plugin? interesting)

**Comment by windmgc** [comment]
Add an action step in `Lint, Doc and Unit tests` to check spec files spelling.

Check for a preview result here: https://github.com/Kong/kong/actions/runs/3640443596/jobs/6145235361

File: scripts/check_spec_files_spelling.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  echo
+  red "----------------------------------------------------------------"
+  echo "Found some files in spec directory that do not have the _spec suffix, please check if you're misspelling them. If there is an exception, please add the coressponding files(or their path regexes) into the whitelist spec/on_demand_specs."
+  echo
+  echo "Possible misspelling file list:"
+  cat $RESULT_TMP
+  red "----------------------------------------------------------------"
+fi
+
+exit $RET
```
Comment (flrgh) [inline scripts/check_spec_files_spelling.sh:38]: I have a suggestion for how to express this list/filter logic in a simpler way (no loops or temp files) while (I hope) still being plenty readable:

```suggestion
readarray -t FOUND < \
<(
  git ls-files 'spec/[0-9]**.lua' \
  | grep -vE \
    -e '_spec.lua$' \
    -f spec/on_demand_specs
)

if (( ${#FOUND[@]} > 0 )); then
  echo
  red "----------------------------------------------------------------"
  echo "Found some files in spec directory that do not have the _spec suffix, please check if you're misspelling them. If there is an exception, please add the coressponding files(or their path regexes) into the whitelist spec/on_demand_specs."
  echo
  echo "Possible misspelling file list:"
  echo
  printf "%s\n" "${FOUND[@]}"
  red "----------------------------------------------------------------"
  exit 1
fi
```

Comment (windmgc) [reply scripts/check_spec_files_spelling.sh]: Very tasty snippet thanks! Tested locally, seems working well

**Review by flrgh — APPROVED** [review]
:+1: lgtm, always appreciate more tests :)

**Comment by flrgh** [comment]
Not merging quite this moment because CI is currently :x: for `master`, but as far as I'm concerned it's good to merge once that is taken care of.

**Pull request merged**